### PR TITLE
Added the disable Minimap feature

### PR DESCRIPTION
**Pull request by henrymkw**

### CODE REVIEW

File: assets/message/MenuSP_U.bmg.json5 [hunk 1]
```diff
@@ -1258,6 +1258,26 @@
...
+        "font": "regular",
+        "string": "Disable",
+    },
+    "10357": {
+        "font": "regular",
+        "string": "Enable",
+    },
+    "10358": {
+        "font": "regular",
+        "string": "MiniMap will be disabled",
```
Comment (riidefi) [inline assets/message/MenuSP_U.bmg.json5:1275]: Style: Usually written "Minimap"

**Comment by stblr** [comment]
I would suggest to set the visibility in `CtrlRace2DMap::calcSelf` instead as this would allow for dynamic switching.

File: payload/game/ui/ctrl/CtrlRace2DMap.cc [hunk 2]
```diff
@@ -3,9 +3,24 @@
...
+#include "game/system/SaveManager.hh"
+
+#include <sp/settings/ClientSettings.hh>
 
 namespace UI {
 
+void CtrlRace2DMap::calcSelf() {
+    auto *saveManager = System::SaveManager::Instance();
+    auto setting = saveManager->getSetting<SP::ClientSettings::Setting::MiniMap>();
+    if (setting == SP::ClientSettings::MiniMap::Disable) {
```
Comment (stblr) [inline payload/game/ui/ctrl/CtrlRace2DMap.cc:15]: You can directly pass the condition to `setVisible`. It also looks to be inverted here.

**Comment by stblr** [comment]
I cannot merge because there are still conflicts in `MenuSP_U.bmg.json5`. Please also run clang-format on `CtrlRace2DMap.cc`.

**Review by stblr — APPROVED** [review]

**Comment by KooShnoo** [comment]
closes #389 ?

**Pull request merged**